src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -114,42 +166,93 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string template
             {
-                OriginalTemplate = JObject.Parse(template),
+                OriginalTemplate = JObject.Parse(populatedTemplate),
                 ExpandedTemplate = templatejObject,
-                IsMainTemplate = true,
+                IsMainTemplate = parentContext.ParentContext == null,
                 ResourceMappings = armTemplateProcessor.ResourceMappings,
                 TemplateIdentifier = templateFilePath,
-                IsBicep = isBicep,
-                SourceMap = sourceMap
+                IsBicep = parentContext.IsBicep,
+                SourceMap = parentContext.SourceMap,
+                PathPrefix = pathPrefix,
+                ParentContext = parentContext
             };
 
             try
             {
                 IEnumerable<IEvaluation> evaluations = this.jsonRuleEngine.AnalyzeTemplate(templateContext);
                 evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
 
-                // For each rule we don't want to report the same line more than once
-                // This is a temporal fix
-                var evalsToValidate = new List<IEvaluation>();
-                var evalsToNotValidate = new List<IEvaluation>();
-                foreach (var eval in evaluations)
+                // Recursively handle nested templates 
+                var jsonTemplate = JObject.Parse(populatedTemplate);
+                var processedTemplateResources = templatejObject.InsensitiveToken("resources");
+
+
+                for (int i = 0; i < processedTemplateResources.Count(); i++)
                 {
-                    if (!eval.Passed && eval.Result != null)
-                    {
-                        evalsToValidate.Add(eval);
-                    }
-                    else
+                    var currentProcessedResource = processedTemplateResources[i];
+
+                    if (currentProcessedResource.InsensitiveToken("type").ToString().Equals("Microsoft.Resources/deployments", StringComparison.OrdinalIgnoreCase))
                     {
-                        evalsToNotValidate.Add(eval);
+                        var nestedTemplate = currentProcessedResource.InsensitiveToken("properties.template");
+                        if (nestedTemplate == null)
+                        {
+                            continue; // This is a Linked template 
+                        }
+                        var populatedNestedTemplate = nestedTemplate.DeepClone();
+
+                        // Check whether scope is set to inner or outer
+                        var scope = currentProcessedResource.InsensitiveToken("properties.expressionEvaluationOptions")?.InsensitiveToken("scope")?.ToString();
+                        //string nextPathPrefix = $".properties.template.";
+                        string nextPathPrefix = nestedTemplate.Path;
+                       
+                        IEnumerable<IEvaluation> result;
```

Comment:
This isn't used until the end.  Move this line to where `result` is assigned and create the variable inline.